co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureRegistrationReceiver.java

```diff
@@ -44,7 +44,7 @@ public class InfrastructureRegistrationReceiver implements Runnable {
     private Queue<InfrastructureRegistrationMessage> rxQueue = new LinkedList<>();
     private DatagramSocket listenSocket = null;
     private static final int listenPort = 1615;
-    private boolean running = false;
+    private AtomicBoolean running = new AtomicBoolean(false);
```

Comment:
Since this is a runnable in which stop can be called outside the running thread, the running boolean flag should be thread safe against concurrent modification.